src/shared/bpf-program.c

```diff
@@ -36,6 +36,8 @@ static const char *const bpf_cgroup_attach_type_table[__MAX_BPF_ATTACH_TYPE] = {
 
 DEFINE_STRING_TABLE_LOOKUP(bpf_cgroup_attach_type, int);
 
+DEFINE_HASH_OPS_WITH_VALUE_DESTRUCTOR(bpf_program_hash_ops, void, trivial_hash_func, trivial_compare_func, BPFProgram, bpf_program_unref);
```

Comment:
This looks strange. The hash op is used for `Set`, not `Hashmap`. So, `KEY_DESTRUCTOR` macro should be used, instead of `VALUE_DESTRUCTOR`.